src/bidiMapper/domains/context/browsingContextProcessor.ts

```diff
@@ -291,15 +291,29 @@ export class BrowsingContextProcessor {
     return context.getOrCreateSandbox(target.sandbox);
   }
 
-  // eslint-disable-next-line @typescript-eslint/require-await
   async process_script_addPreloadScript(
-    _params: Script.AddPreloadScriptParameters
+    params: Script.AddPreloadScriptParameters
   ): Promise<Script.AddPreloadScriptResult> {
-    throw new Message.UnknownErrorException('Not implemented.');
+    const contexts: BrowsingContextImpl[] = [];
+    const scripts: Script.AddPreloadScriptResult[] = [];
 
-    return {
-      script: '', // TODO(#293): Populate script.
-    };
+    if (params.context) {
+      contexts.push(
+        this.#browsingContextStorage.getKnownContext(params.context)
```

Comment:
Here is one edge case with OOPiF. When a frame is moved out of process, we have to add those scripts as well. Can be covered later, but I want to make sure we don't forget about it.